src/main/java/org/sqlite/jdbc3/JDBC3DatabaseMetaData.java

```diff
@@ -1526,7 +1536,7 @@ public ResultSet getIndexInfo(String c, String s, String table, boolean u, boole
         // define the column header
         // this is from the JDBC spec, it is part of the driver protocol
         sql.append("select null as TABLE_CAT,")
-                .append(quote(s))
+                .append(quote(s == null ? "main" : s))
```

Comment:
similar as above, we should return the actual schema of the retrieved PK's table, not the parameter that was passed to the function